File: Eye-openingJava/src/main/java/com/example/eye_openingJava/ui/common/ui/ShareDialogFragment.java
package com.example.eye_openingJava.ui.common.ui;

import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.example.eye_openingJava.R;
import com.example.eye_openingJava.databinding.FragmentShareDialogBinding;
import com.example.eye_openingJava.extension.GlobalExt;
import com.example.eye_openingJava.extension.TextViewExt;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.umeng.analytics.MobclickAgent;

public class ShareDialogFragment extends BottomSheetDialogFragment {

    private FragmentShareDialogBinding binding;

    private String shareContent;
    private Activity attachedActivity;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentShareDialogBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        Activity activity = getActivity();
        if (activity != null) {
            attachedActivity = activity;
            TextViewExt.setDrawable(binding.tvToWechatFriends,ContextCompat.getDrawable(activity, R.drawable.ic_share_wechat_black_30dp), 30f, 30f, 1);
            TextViewExt.setDrawable(binding.tvShareToWeibo,ContextCompat.getDrawable(activity, R.drawable.ic_share_weibo_black_30dp), 30f, 30f, 1);
            TextViewExt.setDrawable(binding.tvShareToQQ,ContextCompat.getDrawable(activity, R.drawable.ic_share_qq_black_30dp), 30f, 30f, 1);
            TextViewExt.setDrawable(binding.tvShareToQQzone,ContextCompat.getDrawable(activity, R.drawable.ic_share_qq_zone_black_30dp), 30f, 30f, 1);

            binding.tvShareToQQ.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    GlobalExt.share(attachedActivity, shareContent, SHARE_QQ);
                    dismiss();
                }
            });
            binding.tvToWechatFriends.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    GlobalExt.share(attachedActivity, shareContent, SHARE_WECHAT);
                    dismiss();
                }
            });
            binding.tvShareToWeibo.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    GlobalExt.share(attachedActivity, shareContent, SHARE_WEIBO);
                    dismiss();
                }
            });
            binding.tvShareToQQzone.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    GlobalExt.share(attachedActivity, shareContent, SHARE_QQZONE);
                    dismiss();
                }
            });
            binding.llMore.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    GlobalExt.share(attachedActivity, shareContent, SHARE_MORE);
                    dismiss();
                }
            });
            binding.tvCancel.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    dismiss();
                }
            });
        }
    }

    public void showDialog(AppCompatActivity activity, String shareContent) {
        if (shareContent.contains(WebViewActivity.DEFAULT_URL)) {
            MobclickAgent.onEvent(activity, Const.Mobclick.EVENT1);
        } else {
            MobclickAgent.onEvent(activity, Const.Mobclick.EVENT2);
        }
        show(activity.getSupportFragmentManager(), "share_dialog");
        this.shareContent = shareContent;
    }

    // 请根据您的需求定义这些常量
    private static final int SHARE_QQ = 1;
    private static final int SHARE_WECHAT = 2;
    private static final int SHARE_WEIBO = 3;
    private static final int SHARE_QQZONE = 4;
    private static final int SHARE_MORE = 0;
}

File: Eye-openingKotlin/src/main/java/com/example/eye_openingKotlin/ui/common/ui/ShareDialogFragment.kt
package com.example.eye_openingKotlin.ui.common.ui

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.example.eye_openingKotlin.Const
import com.example.eye_openingKotlin.R
import com.example.eye_openingKotlin.databinding.FragmentShareDialogBinding
import com.example.eye_openingKotlin.extension.setDrawable
import com.example.eye_openingKotlin.extension.share
import com.example.eye_openingKotlin.util.*
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.umeng.analytics.MobclickAgent


/**
 * 分享对话框的弹出界面。
 *
 * @author vipyinzhiwei
 * @since  2020/5/24
 */
class ShareDialogFragment : BottomSheetDialogFragment() {

    private var _binding: FragmentShareDialogBinding? = null

    private val binding
        get() = _binding!!


    private lateinit var shareContent: String

    private lateinit var attachedActivity: Activity

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = FragmentShareDialogBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        activity?.let {
            attachedActivity = it
            binding.tvToWechatFriends.setDrawable(ContextCompat.getDrawable(it, R.drawable.ic_share_wechat_black_30dp), 30f, 30f, 1)
            binding.tvShareToWeibo.setDrawable(ContextCompat.getDrawable(it, R.drawable.ic_share_weibo_black_30dp), 30f, 30f, 1)
            binding.tvShareToQQ.setDrawable(ContextCompat.getDrawable(it, R.drawable.ic_share_qq_black_30dp), 30f, 30f, 1)
            binding.tvShareToQQzone.setDrawable(ContextCompat.getDrawable(it, R.drawable.ic_share_qq_zone_black_30dp), 30f, 30f, 1)

            binding.tvShareToQQ.setOnClickListener {
                share(attachedActivity, shareContent, SHARE_QQ)
                dismiss()
            }
            binding.tvToWechatFriends.setOnClickListener {
                share(attachedActivity, shareContent, SHARE_WECHAT)
                dismiss()
            }
            binding.tvShareToWeibo.setOnClickListener {
                share(attachedActivity, shareContent, SHARE_WEIBO)
                dismiss()
            }
            binding.tvShareToQQzone.setOnClickListener {
                share(attachedActivity, shareContent, SHARE_QQZONE)
                dismiss()
            }
            binding.llMore.setOnClickListener {
                share(attachedActivity, shareContent, SHARE_MORE)
                dismiss()
            }
            binding.tvCancel.setOnClickListener {
                dismiss()
            }
        }
    }

    fun showDialog(activity: AppCompatActivity, shareContent: String) {
        if (shareContent.contains(WebViewActivity.DEFAULT_URL)) {
            MobclickAgent.onEvent(activity, Const.Mobclick.EVENT1)
        } else {
            MobclickAgent.onEvent(activity, Const.Mobclick.EVENT2)
        }
        show(activity.supportFragmentManager, "share_dialog")
        this.shareContent = shareContent
    }
}